addons/account/models/account_reconcile_model.py

```diff
@@ -521,7 +521,17 @@ def _get_invoice_matching_query(self, st_lines, excluded_ids=None, partner_map=N
             LEFT JOIN account_account account       ON account.id = aml.account_id
             WHERE st_line.id IN %s
                 AND aml.company_id = st_line.company_id
-                AND move.state = 'posted'
+                AND (
+                        -- black line
+                        move.state = 'posted'
+                        OR
+                        -- blue line
+                        (
+                            (move.state = 'draft' OR move.state IS NULL)
+                            AND
+                            journal.post_at = 'bank_rec'
+                        )
+                    )
```

Comment:
Move this diff above:
https://github.com/odoo/odoo/pull/59157/files#diff-dcd37154076abb8655177a7977559898R609
In -- blue lines appearance conditions / -- black lines appearance conditions sections

Also, you can remove `OR move.state IS NULL` since `state` should never be empty